SpinCore_pp/ppg/field_sweep.py

```diff
@@ -0,0 +1,67 @@
+from .. import configureTX,configureRX, configureRX, init_ppg, stop_ppg, runBoard, getData
+from .. import load as spincore_load
+from pyspecdata import *
+from numpy import *
+import time
+def run_field_sweep(B0_idx, indirect_len, nScans, adcOffset, carrierFreq_MHz, 
+        nPoints, SW_kHz, nEchoes, tau_us, p90_us, repetition_us, output_name, 
+        powers, ph1_cyc = r_[0,1,2,3],ph2_cyc = r_[0], sweep_data = None):
+        print("About to run run_scans for", B0_idx)
+        deadtime = 10.0
+        deblank_us = 1.0
+        amplitude = 1.0
+        tx_phases = r_[0.0,90.0,180.0,270.0]
+        nPhaseSteps = len(ph1_cyc)*len(ph2_cyc)
+        total_pts = nPoints*nPhaseSteps
+        data_length = 2*nPoints*nEchoes*nPhaseSteps
+        for x in range(nScans):
+                logger.debug("CONFIGURING TRANSMITTER...")
+                configureTX(adcOffset, carrierFreq_MHz, tx_phases, amplitude, nPoints)
+                logger.debug("\nTRANSMITTER CONFIGURED.")
+                logger.debug("***")
+                logger.debug("CONFIGURING RECEIVER...")
+                acq_time = configureRX(SW_kHz, nPoints, nScans, nEchoes, nPhaseSteps)                
+                logger.debug("\nRECEIVER CONFIGURED.")
+                logger.debug("***")
+                logger.debug("\nINITIALIZING PROG BOARD...\n")
+                init_ppg();
+                logger.debug("\nLOADING PULSE PROG...\n")
+                spincore_load([
+                    ('marker','start',1),
+                    ('phase_reset',1),
+                    ('delay_TTL',deblank_us),
+                    ('pulse_TTL',p90_us,'ph1',ph1_cyc),
+                    ('delay',tau_us),
+                    ('delay_TTL',deblank_us),
+                    ('pulse_TTL',2.0*p90_us,'ph2',ph2_cyc),
+                    ('delay',deadtime),
+                    ('acquire',acq_time),
+                    ('delay',repetition_us),
+                    ('jumpto','start')
```

Comment:
remember to replace this with `run_spin_echo`!!

Reply:
and test it!

Reply:
field sweep runs nicely!